Content.Server/Atmos/Components/InternalsDelayedActivationComponent.cs

```diff
@@ -0,0 +1,20 @@
+﻿namespace Content.Server.Atmos.Components;
+
+/// <summary>
+/// This entity will try to toggle their internals at the specified time
+/// This component is automatically created and deleted by InternalsSystem
+/// </summary>
+[RegisterComponent]
+public sealed partial class InternalsDelayedActivationComponent : Component
+{
+    /// <summary>
+    /// The entity that will toggle internals
+    /// </summary>
+    public EntityUid Entity = new EntityUid();
+
+    /// <summary>
+    /// The server time when the internals will be toggled
+    /// </summary>
+    public TimeSpan Time = new TimeSpan();
+}
+
```

Comment:
You shouldn't be constructing the structs here, also no prediction or datafields which are required.

Reply:
What does prediction do on a server component? I thought that's for the client

Reply:
it should be in shared and predicted

Reply:
Got it, thanks